application/common/models/Reset.php

```diff
@@ -188,16 +186,59 @@ public function sendMethod()
 
     public function sendEmail()
     {
-        /**
-         * @todo send email to $this->method->value with reset code
+        /*
+         * send email to $this->method->value with reset code
          */
     }
 
+    /**
+     * Send phone verification and store resulting code
+     * @throws \Exception
+     */
     public function sendPhone()
     {
-        /**
-         * @todo use phone service component to send notification to $this->method->value
-         */
+        // Initialize log
+        $log = [
+            'action' => 'reset',
+            'method' => 'phone',
+            'method_id' => $this->method_id,
+            'previous_attempts' => $this->attempts,
+        ];
+
+        // Get phone number without comma
+        $number = $this->method->getRawPhoneNumber();
+
+        // Generate random code for potential use
+        $code = Utils::getRandomDigits(\Yii::$app->phone->codeLength);
```

Comment:
Should we add something to the PhoneVerificationInterface to make sure that every implementation will have a `codeLength` property?